Library/Homebrew/formula_creator.rb

```diff
@@ -82,7 +82,10 @@ def write_formula!
           r = Resource.new
           r.url(@url)
           r.owner = self
-          @sha256 = r.fetch.sha256 if r.download_strategy == CurlDownloadStrategy
+          filepath = r.fetch
+          raise "Downloaded URL is not archive" if File.read(filepath, 100).strip.start_with?("<!DOCTYPE html>")
```

Comment:
This seems like it is overly broad. I think it'd be worth checking the URL and recommending the correct one instead.

Reply:
Ok, thanks. Why `100` here? would make more sense to be the length of the specific string?

Reply:
I didn't know what to check. 100 seems to be a fine length to detect different file signatures without slurping the whole file into memory. For example, to detect that the file is actual archive. But it appeared easier to just spot what is 100% not expected here.

Reply:
```suggestion
          html_doctype_prefix = "<!doctype html"
          raise "Downloaded URL is not archive" if File.read(filepath, html_doctype_prefix.length).downcase.start_with?(html_doctype_prefix)
```
something like this would make more sense to me to avoid reading things we don't need to and being more robust on the specific doctype

Reply:
Thanks for the review. That's more readable, indeed. Committed the suggestion.

Reply:
@MikeMcQuaid actually this doesn't work in case doctype is prefixed by spaces. The fix is to get back `.strip` and fixed probe length https://github.com/Homebrew/brew/pull/20026